src/main/java/edu/pw/eiti/pik/user/UserController.java

```diff
@@ -47,6 +57,25 @@ UserDto me() {
         return userMapper.toDto(user);
     }
 
+    @ApiOperation(value = "Get top n users by name who have given role")
+    @ApiResponses({
+    	@ApiResponse(code = 200, message = "Always")
+    })
+    @GetMapping(path = "/user/{authority}/{name}/{n}")
+    List<UserDto> getUsersByRoleAndName(
+    		@PathVariable String authority,
+    		@PathVariable String name,
+    		@PathVariable int n)
+    {
+    	Page<User> res = userESRepository.findByNameAndAuthorityName(
```

Comment:
All the operations on data model should be wrapped by service. In multilayer architecture controller invokes services, services work on data model. Check out ParticipationController and ParticipationService or even login in UserController and UserService